python/cuda_parallel/cuda/parallel/experimental/algorithms/reduce.py

```diff
@@ -42,12 +45,18 @@ def _dtype_validation(dt1, dt2):
 
 class _Reduce:
     # TODO: constructor shouldn't require concrete `d_in`, `d_out`:
-    def __init__(self, d_in, d_out, op: Callable, h_init: np.ndarray):
+    def __init__(
+        self,
+        d_in: DeviceArrayLike | IteratorBase,
```

Comment:
Note to self: Python 3.7 isn't going to like this.

Reply:
Instead, I used `from __future__ import annotations` which will make migrating easy.